mne/chpi.py

```diff
@@ -295,14 +295,51 @@ def extract_chpi_locs_kit(raw, stim_channel='MISC 064', *, verbose=None):
 
 # ############################################################################
 # Estimate positions from data
+
 @verbose
-def _get_hpi_info(info, allow_empty=False, verbose=None):
-    """Get HPI information from raw."""
+def get_chpi_info(inst, on_missing='ignore', verbose=None):
+    """Retrieve cHPI information from the data.
+
+    Parameters
+    ----------
+    inst : instance of Info | Raw | Epochs | Evoked
+        The object from which to extract the HPI information.
+    on_missing : 'ignore' | 'raise'
+        What to do if no HPI information can be found. If ``'ignore'``
+        (default), all return values will be empty arrays or ``None``. If
+        ``'raise'``, an exception will be raised.
+    %(verbose)s
+
+    Returns
+    -------
+    hpi_freqs : array, shape (n_coils,)
+        The frequency used for each individual cHPI coil.
+    hpi_pick : int | None
+        The index of the ``STIM`` channel containing information about when
+        which cHPI coils were switched on.
+    hpi_on : array, shape (n_coils,)
+        The values coding for the "on" state of each individual cHPI coil.
+
+    Notes
+    -----
+    .. versionadded:: 0.24
+    """
+    from .io import BaseRaw
+    from . import BaseEpochs
+    from .evoked import Evoked
+
+    _validate_type(item=inst, item_name='inst',
+                   types=(Info, BaseRaw, BaseEpochs, Evoked))
+    _check_option(parameter='on_missing', value=on_missing,
+                  allowed_values=['ignore', 'raise'])
+
+    info = inst if isinstance(inst, Info) else inst.info
```

Comment:
I think I would rather just have people pass `inst.info` here

Reply:
I thought supporting more object types could provide slightly more convenience for users … no? :)

Reply:
I would stick to passing info.

opening more options means more tests, more documentation etc.